hoodie-client/src/test/java/com/uber/hoodie/table/TestMergeOnReadTable.java

```diff
@@ -564,6 +567,186 @@ public void testRollbackWithDeltaAndCompactionCommit() throws Exception {
+    records.addAll(newInserts);
+    writeRecords = jsc.parallelize(records, 1);
+
+    writeStatusJavaRDD = client.upsert(writeRecords, newCommitTime);
+    client.commit(newCommitTime, writeStatusJavaRDD);
+    statuses = writeStatusJavaRDD.collect();
+    // Verify there are no errors
+    assertNoWriteErrors(statuses);
+
+    metaClient = new HoodieTableMetaClient(jsc.hadoopConfiguration(), cfg.getBasePath());
+
+    String compactionInstantTime = "004";
+    allCommits.add(compactionInstantTime);
+    client.scheduleCompactionAtInstant(compactionInstantTime, Optional.empty());
+
+    // Compaction commit
+    /**
+     * Write 4 (updates)
+     */
+    newCommitTime = "005";
+    allCommits.add(newCommitTime);
+    client.startCommitWithTime(newCommitTime);
+
+    records = dataGen.generateUpdates(newCommitTime, records);
+    writeRecords = jsc.parallelize(records, 1);
+
+    writeStatusJavaRDD = client.upsert(writeRecords, newCommitTime);
+    client.commit(newCommitTime, writeStatusJavaRDD);
+    statuses = writeStatusJavaRDD.collect();
+    // Verify there are no errors
+    assertNoWriteErrors(statuses);
+
+    metaClient = new HoodieTableMetaClient(jsc.hadoopConfiguration(), cfg.getBasePath());
+
+    compactionInstantTime = "006";
+    allCommits.add(compactionInstantTime);
+    client.scheduleCompactionAtInstant(compactionInstantTime, Optional.empty());
+    JavaRDD<WriteStatus> ws = client.compact(compactionInstantTime);
+    client.commitCompaction(compactionInstantTime, ws, Optional.empty());
+
+    allFiles = HoodieTestUtils.listAllDataFilesInPath(metaClient.getFs(), cfg.getBasePath());
+    metaClient = new HoodieTableMetaClient(jsc.hadoopConfiguration(), cfg.getBasePath());
+    roView = new HoodieTableFileSystemView(metaClient, metaClient.getCommitsTimeline(), allFiles);
+
+    final String compactedCommitTime = metaClient.getActiveTimeline().reload().getCommitsTimeline().lastInstant().get()
+        .getTimestamp();
+
+    assertTrue(roView.getLatestDataFiles().filter(file -> {
+      if (compactedCommitTime.equals(file.getCommitTime())) {
+        return true;
+      } else {
+        return false;
+      }
+    }).findAny().isPresent());
+
+    /**
+     * Write 5 (updates)
+     */
+    newCommitTime = "007";
+    allCommits.add(newCommitTime);
+    client.startCommitWithTime(newCommitTime);
+    copyOfRecords = new ArrayList<>(records);
+    copyOfRecords = dataGen.generateUpdates(newCommitTime, copyOfRecords);
+    copyOfRecords.addAll(dataGen.generateInserts(newCommitTime, 200));
+
+    statuses = client.upsert(jsc.parallelize(copyOfRecords, 1), newCommitTime).collect();
+    // Verify there are no errors
+    assertNoWriteErrors(statuses);
+    client.commit(newCommitTime, writeStatusJavaRDD);
+    copyOfRecords.clear();
+
+    // Rollback latest commit first
+    allCommits.stream().sorted(Comparator.reverseOrder()).forEach(commitToRollback -> {
+      client.rollback(commitToRollback);
```

Comment:
Call client.restoreToCommit() here instead of rollback ?  Will improve coverage too in HoodieWriteClient.

Reply:
+1